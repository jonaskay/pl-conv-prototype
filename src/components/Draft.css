.Draft {
  background: #303030; 
  box-shadow:
    0 1px 5px 0 rgba(0,0,0,.2), 0 2px 2px 0 rgba(0,0,0,.14), 0 3px 1px -2px rgba(0,0,0,.12);
  display: flex;
  height: 100%;
  position: relative;
  width: 100%;
}

.Draft-col1 {
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 60%;
}

.Draft-col2 {
  border-left: 1px solid rgba(255,255,255,.12);
  height: 100%;
  overflow-y: auto;
  width: 40%;
}

.Draft-header {
  flex: none;
  margin-bottom: 1rem;
  padding: 1rem;
}

.Draft-table {
  border-collapse: collapse;
  display: flex;
  flex: auto;
  flex-direction: column;
  height: 100%;
  width: 100%;
}

.Draft-table-header {
  border-bottom: 1px solid rgba(255,255,255,.12);
  display: flex;
  flex: none;
}

.Draft-table-header > div {
  box-sizing: border-box;
  color: rgba(255,255,255,.5);
  font-size: .875rem;
  font-weight: 300;
  letter-spacing: .05em;
  padding: .75rem 1.5rem;
  padding-right: 0;
  text-transform: uppercase;
}

.Draft-table-body {
  flex: auto;
  overflow-y: auto;
}
