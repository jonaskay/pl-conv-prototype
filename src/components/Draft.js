import React from 'react';
import './Draft.css';
import DraftNameInput from './DraftNameInput';
import DraftItemCounter from './DraftItemCounter';
import DraftItem from './DraftItem';
import Track from './Track';

class Draft extends React.Component {
  render() {
    return (
      <div className="Draft">
        <div className="Draft-col1">
          <div className="Draft-header">
            <div style={{color: 'rgba(255,255,255,.5)'}}>
              {this.props.name} &#9642; <DraftItemCounter items={this.props.items} />
            </div>
            <DraftNameInput name={this.props.name} />
          </div>          
          <div class="Draft-table">
            <div class="Draft-table-header">
              <div style={{flex: 'none', width: '13.5rem'}}>Name</div>
              <div style={{flex: 'auto'}}>Artist</div>
            </div>
            <div class="Draft-table-body">
              {this.props.items.map(
                  (item, i) => <DraftItem key={item.id} item={item} selected={i === 0} />
              )}
            </div>
          </div>
        </div>
        <div className="Draft-col2">
          <Track track={this.props.items[0]} />
        </div>
      </div>
    );
  }
}

export default Draft;
