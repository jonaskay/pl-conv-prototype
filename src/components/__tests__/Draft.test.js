import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Draft from '../Draft';

const item = { 
  "id": 1,
  "name": "Bodak Yellow", 
  "artist": "Cardi B",
  "image": "https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc",
  "spotify_id": "2771LMNxwf62FTAdpJMQfM",
  "youtube_id": "PEGccV-NOm8"
}

it('renders a playlist draft', () => {
  const component = ReactTestRenderer.create(<Draft items={[item]} />);
  expect(component).toMatchSnapshot();
});
