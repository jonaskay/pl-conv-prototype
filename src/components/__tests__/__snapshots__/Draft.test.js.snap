// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a playlist draft 1`] = `
<div
  className="Draft"
>
  <div
    className="Draft-col1"
  >
    <div
      className="Draft-header"
    >
      <div
        style={
          Object {
            "color": "rgba(255,255,255,.5)",
          }
        }
      >
         â–ª 
        <span>
          1
           
          track
        </span>
      </div>
      <input
        className="DraftNameInput"
        type="text"
        value={undefined}
      />
    </div>
    <div
      class="Draft-table"
    >
      <div
        class="Draft-table-header"
      >
        <div
          style={
            Object {
              "flex": "none",
              "width": "13.5rem",
            }
          }
        >
          Name
        </div>
        <div
          style={
            Object {
              "flex": "auto",
            }
          }
        >
          Artist
        </div>
      </div>
      <div
        class="Draft-table-body"
      >
        <div
          className="DraftItem DraftItem-selected"
        >
          <div
            style={
              Object {
                "flex": "none",
                "width": "13.5rem",
              }
            }
          >
            Bodak Yellow
          </div>
          <div>
            Cardi B
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Draft-col2"
  >
    <div
      className="Track"
    >
      <div
        className="Track-backdrop"
        style={
          Object {
            "backgroundImage": "url(https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc)",
          }
        }
      />
      <div
        className="Track-content"
      >
        <div
          style={
            Object {
              "marginBottom": "1.25rem",
            }
          }
        >
          <img
            alt="Bodak Yellow - Cardi B"
            className="Track-img"
            src="https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc"
          />
          <div
            style={
              Object {
                "fontSize": "1.125rem",
                "marginBottom": ".25rem",
              }
            }
          >
            Bodak Yellow
          </div>
          <div>
            Cardi B
          </div>
        </div>
        <div
          style={
            Object {
              "marginBottom": "1rem",
            }
          }
        >
          <iframe
            frameborder="0"
            height="80"
            src="https://open.spotify.com/embed?uri=spotify:track:2771LMNxwf62FTAdpJMQfM"
            title="Spotify Play Button"
            width="288"
          />
        </div>
        <div
          style={
            Object {
              "marginBottom": "1rem",
            }
          }
        >
          <iframe
            allow="autoplay; encrypted-media"
            allowFullScreen={true}
            frameborder="0"
            height="162"
            src="https://www.youtube.com/embed/PEGccV-NOm8?controls=0&amp;showinfo=0"
            title="YouTube Player"
            width="288"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
